src/sonic-py-common/sonic_py_common/device_info.py

```diff
@@ -256,7 +257,11 @@ def get_path_to_port_config_file(hwsku=None, asic=None):
     # if 'hwsku.json' file is available, Check for 'platform.json' file presence,
     # if 'platform.json' is available, APPEND it. Otherwise, SKIP it.
     if os.path.isfile(hwsku_json_file):
-        port_config_candidates.append(os.path.join(platform_path, PLATFORM_JSON_FILE))
+        if os.path.isfile(os.path.join(platform_path, PLATFORM_JSON_FILE)):
+            json_file = os.path.join(platform_path, PLATFORM_JSON_FILE)
+            platform_data = json.loads(open(json_file).read())
+            if len(platform_data.get("interfaces")) > 0:
```

Comment:
Suggest adding a comment, stating that we only consider platform.json to be the port configuration file if it contains actual interface details. I believe we intend to eventually make this mandatory and transition away from the old port_config.ini file, so maybe also add a TODO to eventually remove the check.